src/App.tsx

```diff
@@ -1,9 +1,22 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const lowerQuery = query.toLowerCase();
+
+  const handlerQueryChange = (value: string) => {
+    setQuery(value);
+  };
+
+  const visibleMovies = moviesFromServer
+    .filter(movie => {
+      return movie.title.toLowerCase().includes(lowerQuery)
+      || movie.description.toLowerCase().includes(lowerQuery);
+    });
```

Comment:
still looks complicated :(
here you could have used destructuring, vars, new separate function like `containsQuery` – everything that could make it look simpler